packages/next/shared/lib/match-remote-pattern.ts

```diff
@@ -0,0 +1,68 @@
+import type { RemotePattern } from './image-config'
+
+export function matchRemotePattern(pattern: RemotePattern, url: URL): boolean {
+  if (pattern.protocol !== undefined) {
+    const actualProto = url.protocol.slice(0, -1)
+    if (pattern.protocol !== actualProto) {
+      return false
+    }
+  }
+  if (pattern.port !== undefined) {
+    if (pattern.port !== url.port) {
+      return false
+    }
+  }
+  if (pattern.pathname !== undefined) {
+    const patternParts = pattern.pathname.split('/')
+    const actualParts = url.pathname.split('/')
+    const len = Math.max(patternParts.length, actualParts.length)
+    for (let i = 0; i < len; i++) {
+      if (patternParts[i] === '**' && actualParts[i] !== undefined) {
+        // Double asterisk means "match everything until the end of the path"
+        // so we can break the loop early
+        break
+      }
+      if (patternParts[i] === '*') {
+        // Single asterisk means "match this part" so we can
+        // continue to the next part of the loop
+        continue
+      }
+      if (patternParts[i] !== actualParts[i]) {
```

Comment:
Do we need to worry about casing here or do we expect it to match exactly?

Reply:
I think exact match is best. Does `rewrites` work with different case?

Reply:
`rewrites` are currently case-insensitive https://github.com/vercel/next.js/blob/d1db7141ba029acbafcec3684e469611c59b43ad/packages/next/shared/lib/router/utils/path-match.ts#L32

Reply:
Interesting, they are sensitive in `vercel.json`. I tend to think `sensitive: true` is the correct default for two reasons:

1. Avoid duplicate content (insensitive would allow 2 urls to serve the same content)
2. Avoid cache invalidation (insensitive would allow multiple cache keys for the same content)